### PR TITLE
traefik: Upgrade to latest upstream traefik chart

### PR DESCRIPTION
**Pull request by gracedo**
**What type of PR is this?**

Bug

**What this PR does/ why we need it**:

Upgrade to latest upstream traefik chart to pull in necessary changes to get metrics scraping working again

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-70005

**Special notes for your reviewer**:

This upgrade was very manual/tedious so I would love some others who are more familiar with this chart to verify that I didn't miss any additions we've made on our fork. I tried to go through the diff of the latest traefik chart and our fork (prior to and post upgrade) and cross referenced the diff against our fork's commit [history](https://github.com/mesosphere/charts/commits/master/staging/traefik) to check what changes to keep.

Here is the diff against upstream traefik chart (as base): https://gist.github.com/gracedo/1d03fd3b43b69faa963425a5875001c6

##Testing
I tested this via `gracedo/traefik_upgrade_D2IQ-70005` [kba branch](https://github.com/mesosphere/kubernetes-base-addons/tree/gracedo/traefik_upgrade_D2IQ-70005) ([pr here](https://github.com/mesosphere/kubernetes-base-addons/pull/349)) which pulls in my self-hosted version of the chart. I verified that prom is successfully scraping traefik metrics, and that they are presented successfully in grafana.
image
image
image

**Does this PR introduce a user-facing change?**:

```release-note
The option `deploymentAnnotation` has been deprecated; please use `deployment.Annotations` instead.
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: staging/traefik/templates/dashboard-ingress.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.dashboard.enabled }}
-apiVersion: networking.k8s.io/v1beta1
+apiVersion: extensions/v1beta1
```
Comment (dkoshkin) [inline staging/traefik/templates/dashboard-ingress.yaml:2]: Please revert

File: staging/traefik/templates/deployment.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {{- if semverCompare "^1.9-0" .Capabilities.KubeVersion.GitVersion }}
 apiVersion: apps/v1
 {{- else }}
-apiVersion: apps/v1
+apiVersion: apps/v1beta1
```
Comment (dkoshkin) [inline staging/traefik/templates/deployment.yaml:4]: Please revert

File: staging/traefik/templates/deployment.yaml [hunk 3]
```diff
@@ -11,12 +11,17 @@ metadata:
     chart: {{ template "traefik.chart" . }}
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
-  {{- if .Values.deploymentAnnotations }}
+    {{- if .Values.deployment.labels }}
+    {{- toYaml .Values.deployment.labels | nindent 4 }}
+    {{- end }}
+  {{- with .Values.deployment.annotations }}
```
Comment (dkoshkin) [inline staging/traefik/templates/deployment.yaml:17]: We should document this change in the release-notes, users might be using this option.

File: staging/traefik/README.md [hunk 4]
```diff
@@ -247,6 +273,8 @@ The following table lists the configurable parameters of the Traefik chart and t
 | `forwardAuth.authResponseHeaders`      | Set authentication response headers                                                                                                  |  `[]`                                     |
 | `initCertJobImage`                     | Set the Docker image to be used for updating the cert-manager certificate with the load balancer SANs                        | `mesosphere/kubeaddons-addon-initializer:v0.0.7`  |
 | `extraConfigEntrypoints`               | Extra configurations for the entrypoints.                                                                                    | None                                              |
+| `extraVolumeMounts`      | Any extra volumes mounts to define for the Traefik container                                                                                                                                                                                                               | `[]`                                                                                                                      |
```
Comment (jieyu) [inline staging/traefik/README.md:276]: not yours, but could you add `extraSSLEntrypoints` here which is something we introduced

File: staging/traefik/templates/deployment.yaml (hunk 3 above)
Comment (gracedo) [reply staging/traefik/templates/deployment.yaml:17]: Added in release notes section

File: staging/traefik/templates/deployment.yaml [hunk 5]
```diff
@@ -161,30 +175,41 @@ spec:
...
           protocol: TCP
         - name: https
-          containerPort: 443
+          containerPort: {{ template "traefik.containerPort.https" . }}
           {{- if .Values.deployment.hostPort.httpsEnabled }}
           hostPort: {{ default 443 .Values.deployment.hostPort.httpsPort }}
           {{- end }}
           protocol: TCP
-        {{- if .Values.dashboard.enabled }}
+        {{- if or .Values.dashboard.enabled .Values.metrics.prometheus.enabled }}
```
Comment (jieyu) [inline staging/traefik/templates/deployment.yaml:183]: hum, this looks weird. is that what upstream is doing as well?

**Comment by gracedo** [comment]
I enabled the service monitor in my kba branch ([here](https://github.com/mesosphere/kubernetes-base-addons/blob/gracedo/traefik_upgrade_D2IQ-70005/addons/traefik/1.7.x/traefik-21.yaml#L73-L80)), but the metrics targets are still down 🤔  looking into this..
image

File: staging/traefik/templates/deployment.yaml (hunk 5 above)
Comment (gracedo) [reply staging/traefik/templates/deployment.yaml:183]: yeah, this is pulled from upstream, looks like it was intro'd [here](https://github.com/helm/charts/pull/15210). doesn't look like that change is needed now that dashboard and prometheus metrics are decoupled, maybe they missed this line. I can revert the line and also open a pr against upstream

Comment (gracedo) [reply staging/traefik/templates/deployment.yaml]: https://github.com/helm/charts/pull/23156

**Comment by gracedo** [comment]
This is working now with the latest fixes! Please see the description for screenshots of the traefik targets successfully up in prom, and the grafana dashboard displaying metrics 🎉

File: staging/traefik/ci/ci-values.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+serviceType: NodePort
+testFramework:
+  enabled: true
```
Comment (sebbrandt87) [inline staging/traefik/ci/ci-values.yaml:3]: Please drop the whole file. It breaks our tests.

**Comment by sebbrandt87** [comment]
just added a test local and dropped the `staging/traefik/ci/ci-values.yaml` file.
will push into your branch, if I can, otherwise I will send you a diff file

**Comment by sebbrandt87** [comment]
As you see in my commit that I just pushed, as I had time now, we also need to update the `deployment.annotations` part of kubernetes-base-addons later as otherwise no reload will be triggered by reloader for the traefik deployment.
At best also the changes I did for e2e-test.sh are also still working within a started container in a cluster. If not I will revert that part, but we definitely need to make more use of tests also running local :)

File: staging/traefik/ci/ci-values.yaml (hunk 6 above)
Comment (sebbrandt87) [reply staging/traefik/ci/ci-values.yaml]: removed

**Comment by sebbrandt87** [comment]
```------------------------------------------------------------------------------------------------------------------------
==> Logs of container traefik-x1v5lavujl-test
------------------------------------------------------------------------------------------------------------------------
>>> kubectl logs traefik-x1v5lavujl-test --namespace traefik-x1v5lavujl --container traefik-x1v5lavujl-test
1..6
ok 1 Test HTTP
ok 2 Test HTTPS
ok 3 Test Default Certificate
ok 4 Test Default Certificate SNI
ok 5 Test Custom Certificate SNI
ok 6 Test Prometheus Metrics scrape
------------------------------------------------------------------------------------------------------------------------
<== Logs of container traefik-x1v5lavujl-test
------------------------------------------------------------------------------------------------------------------------```

**Comment by gracedo** [comment]
thank you @sebbrandt87 !

**Pull request merged**